src/context/cdhashmap.h

```diff
@@ -101,13 +101,26 @@ class CDOhash_map : public ContextObj {
   friend class CDHashMap<Key, Data, HashFcn>;
 
  public:
-  // The type of the <Key, Data> value mapped to by this class.
-  using value_type = std::pair<Key, Data>;
+  // The type of the <Key, Data> pair mapped by this class.
+  //
+  // Implementation:
+  // The data and key visible to users of CDHashMap are only visible through
+  // const references. Thus the type of dereferencing a
+  // CDHashMap<Key, Data>::iterator.second should be value_type::second_type
+  // should be a `const Data&`. (Otherwise, access operations would need to
+  // makeCurrent() to get a Data&, which is an unacceptable performance hit.)
+  // To allow for the desired updating in other context, we store a
+  // std::pair<const Key, const Data> and break the const encapsulation when
```

Comment:
contexts

Reply:
I went with "scenarios" as "contexts" is potentially confusing here.